lib/services/catalog/controllers/catalog-search-controller.js

```diff
@@ -12,20 +13,64 @@ class CatalogSearchController extends Controller {
 	}
 
 	get(req, res, next) {
-		const query = req.query.q;
+		const query = _.get(req, 'query.q', '');
 		const channel = (req.identity.channel || {}).id;
-		const type = req.query.type;
 
 		if (!channel) {
 			return next(Boom.badRequest('JSON Web Token must include a channel ID for search.'));
 		}
 
-		const args = {query, channel, type};
+		if (query.length <= 3) {
+			return next(Boom.badRequest('`q` query paramater must be at least 3 characters'));
+		}
+
+		let types;
+		if ((req.query || {}).types) {
+			// At this time, only allow sorting by one paramater.
+			types = req.query.types.split(',');
+		} else {
+			types = ['video', 'collection'];
```

Comment:
Just realized that if we have defaults, we need to set them on `res.query` when they aren't set, so that the response meta contains the proper results. So if `types` isn't defined at all, we want to do `res.query = Object.assign({}, res.query, {types});` and also `res.query = Object.assign({}, res.query, { sort: sortBy });`